GIData/src/commonMain/kotlin/org/anime_game_servers/game_data_models/gi/data/activity/ActivityType.kt

```diff
@@ -411,49 +426,54 @@ enum class ActivityType(val id: Int) : IntKey {
     NEW_ACTIVITY_MICHIAE_MATSURI(2501),
 
     /**
-     * TODO identify this activity
+     * This is the 2.5 "[Of Drink A-Dreaming](https://genshin-impact.fandom.com/wiki/Of_Drink_A-Dreaming)" event
      * TODO document used models
      */
     NEW_ACTIVITY_BARTENDER(2502),
 
     /**
-     * TODO identify this activity
+     * This is the 2.5 "[Divine Ingenuity](https://genshin-impact.fandom.com/wiki/Divine_Ingenuity/2022-03-02)" event
      * TODO document used models
      */
     NEW_ACTIVITY_UGC(2503),
 
     /**
-     * TODO identify this activity
+     * This is the 2.6 "[Hues of the Violet Garden](https://genshin-impact.fandom.com/wiki/Hues_of_the_Violet_Garden)" event
      * TODO document used models
      */
     NEW_ACTIVITY_IRODORI(2600),
 
     /**
-     * TODO identify this activity
+     * This is the 2.6 "[Vibro-Crystal Research](https://genshin-impact.fandom.com/wiki/Vibro-Crystal/2022-04-21)" event
+     * This is likely the 3.5 "[Vibro-Crystal Verification](https://genshin-impact.fandom.com/wiki/Vibro-Crystal/2023-03-14)" event
+     * This is likely the 4.6 "[Vibro-Crystal Applications](https://genshin-impact.fandom.com/wiki/Vibro-Crystal/2024-04-28)" event
      * TODO document used models
      */
     NEW_ACTIVITY_CRYSTAL_LINK(2601),
 
     /**
-     * TODO identify this activity
+     * This is the 2.6 "[Spices From the West](https://genshin-impact.fandom.com/wiki/Spices_From_the_West/2022-05-14)" event
+     * This is likely the 3.5 "[Spices From the West: Northerly Search](https://genshin-impact.fandom.com/wiki/Spices_From_the_West/2023-03-31)" event
      * TODO document used models
      */
     NEW_ACTIVITY_SPICE(2602),
 
     /**
-     * TODO identify this activity
+     * This is the 2.6 "[Outside the Canvas, Inside the Lens](https://genshin-impact.fandom.com/wiki/Outside_the_Canvas,_Inside_the_Lens/2022-04-01)" event
+     * This is the 3.2 "[Outside the Canvas, Inside the Lens: Greenery Chapter](https://genshin-impact.fandom.com/wiki/Outside_the_Canvas,_Inside_the_Lens/2022-11-16)" event
+     * This is likely the 4.8 "[Outside the Canvas, Inside the Lens: Dew-Kissed Chapter](https://genshin-impact.fandom.com/wiki/Outside_the_Canvas,_Inside_the_Lens/2024-07-29)" event
```

Comment:
some seem to be missing here, since in the 5.4 activity excel it has 5 entries that use it.